homeassistant/components/wallbox/coordinator.py

```diff
@@ -101,47 +119,42 @@ async def async_validate_input(self) -> None:
         """Get new sensor data for Wallbox component."""
         await self.hass.async_add_executor_job(self._validate)
 
+    @_require_authentication
     def _get_data(self) -> dict[str, Any]:
         """Get new sensor data for Wallbox component."""
-        try:
-            self._authenticate()
-            data: dict[str, Any] = self._wallbox.getChargerStatus(self._station)
-            data[CHARGER_MAX_CHARGING_CURRENT_KEY] = data[CHARGER_DATA_KEY][
-                CHARGER_MAX_CHARGING_CURRENT_KEY
-            ]
-            data[CHARGER_LOCKED_UNLOCKED_KEY] = data[CHARGER_DATA_KEY][
-                CHARGER_LOCKED_UNLOCKED_KEY
-            ]
-            data[CHARGER_ENERGY_PRICE_KEY] = data[CHARGER_DATA_KEY][
-                CHARGER_ENERGY_PRICE_KEY
-            ]
-            data[
-                CHARGER_CURRENCY_KEY
-            ] = f"{data[CHARGER_DATA_KEY][CHARGER_CURRENCY_KEY][CODE_KEY]}/kWh"
-
-            data[CHARGER_STATUS_DESCRIPTION_KEY] = CHARGER_STATUS.get(
-                data[CHARGER_STATUS_ID_KEY], ChargerStatus.UNKNOWN
-            )
-            return data
-        except (
-            ConnectionError,
-            requests.exceptions.HTTPError,
-        ) as wallbox_connection_error:
-            raise UpdateFailed from wallbox_connection_error
+        data: dict[str, Any] = self._wallbox.getChargerStatus(self._station)
+        data[CHARGER_MAX_CHARGING_CURRENT_KEY] = data[CHARGER_DATA_KEY][
+            CHARGER_MAX_CHARGING_CURRENT_KEY
+        ]
+        data[CHARGER_LOCKED_UNLOCKED_KEY] = data[CHARGER_DATA_KEY][
+            CHARGER_LOCKED_UNLOCKED_KEY
+        ]
+        data[CHARGER_ENERGY_PRICE_KEY] = data[CHARGER_DATA_KEY][
+            CHARGER_ENERGY_PRICE_KEY
+        ]
+        data[
+            CHARGER_CURRENCY_KEY
+        ] = f"{data[CHARGER_DATA_KEY][CHARGER_CURRENCY_KEY][CODE_KEY]}/kWh"
+
+        data[CHARGER_STATUS_DESCRIPTION_KEY] = CHARGER_STATUS.get(
+            data[CHARGER_STATUS_ID_KEY], ChargerStatus.UNKNOWN
+        )
+        return data
 
     async def _async_update_data(self) -> dict[str, Any]:
         """Get new sensor data for Wallbox component."""
         return await self.hass.async_add_executor_job(self._get_data)
 
+    @_require_authentication
```

Comment:
As you manually call authentication in the function. It makes no sense to use `_require_authentication` too, as this would lead to call authenticate twice

Reply:
yes, true, I think I left that there by accident.